### PR TITLE
Feat/order forms

### PR DESCRIPTION
**Pull request by Sontranwakumo**
### Description

This PR include works:
- Initialize Order forms resources. 
- Add Readme file to describe utilities API `POST /order-forms` create Order forms.
- Add outline of testcases for API `POST /order-forms`.
- Write my first test case at `src/order-forms/order-forms.controller.spec.ts`.
- Write `create order-form` logic at `src/order-forms/order-form.service.ts`.

### CODE REVIEW

File: src/v1/order-forms/order-forms.controller.spec.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    datasource = await getSynchronizeConnection();
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [...IMPORT_MODULES, OrderFormsModule],
+    }).compile();
+    app = module.createNestApplication();
+    await app.init();
+  });
+
+  afterEach(async () => {
+    // await clearDB(datasource);
```
Comment (longhoangwkm) [inline src/v1/order-forms/order-forms.controller.spec.ts]: always enable clean Database for each test.

File: src/v1/order-forms/order-forms.controller.spec.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  });
+  describe('POST order-forms', () => {
```
Comment (longhoangwkm) [inline src/v1/order-forms/order-forms.controller.spec.ts]: Missing version, ex: /v1/order-forms

File: src/v1/order-forms/order-forms.controller.spec.ts [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+      branch.address = 'Le Dinh Ly, Da Nang';
+      await branch.save();
+      //initialize court
+      court = new CourtInfo();
+      court.branch = branch;
+      court.id_branch = branch.id;
+      court.description = 'Sân trong nhà, gần khán đài';
+      court.name_of_court = 'San 1';
+      await court.save();
+    });
```
Comment (longhoangwkm) [inline src/v1/order-forms/order-forms.controller.spec.ts:52]: Move to test/ultil.ts

createRandomUser() function, using `@faker-js/faker` for random data

Comment (longhoangwkm) [reply src/v1/order-forms/order-forms.controller.spec.ts:52]: ex:

```js
const { faker } = require('@faker-js/faker');

function getRandomUser(role = UserRole.USER) {
  const user = new User();
  user.username = faker.internet.userName();
  user.email = faker.internet.email();
  user.password = faker.internet.password();
  user.fullname = faker.name.fullName();
  user.phone = faker.phone.number();
  user.role = role;
  return user;
}
```

File: src/v1/order-forms/order-forms.controller.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import { Controller, Get, Post, Body, Patch, Param, Delete } from '@nestjs/common';
+import { OrderFormsService } from './order-forms.service.js';
+import { CreateOrderFormDto } from './dto/create-order-form.dto.js';
+import { UpdateOrderFormDto } from './dto/update-order-form.dto.js';
+
+@Controller('order-forms')
+export class OrderFormsController {
```
Comment (longhoangwkm) [inline src/v1/order-forms/order-forms.controller.ts:7]: Missing version

File: src/v1/order-forms/order-forms.controller.spec.ts (hunk 3 above)
Comment (longhoangwkm) [reply src/v1/order-forms/order-forms.controller.spec.ts:52]:
```
const { user, admin, branch, court } = await initCourtData();
```

**Comment by Sontranwakumo** [comment]
@longhoangwkm I completed fix controller route to `/v1/order-forms` and implement `faker-js/faker` on generate new object in the latest commit. I also uncomment the `clearDB` row. Thank you

File: src/v1/order-forms/order-forms.controller.spec.ts [hunk 5]
```diff
@@ -93,14 +68,12 @@ describe('OrderFormsController', () => {
...
           it('Should return 404 and an error message for user that does not exist', async () => {
-            // const wdto = instanceToInstance(dto);
-            // wdto.sender_id = null;
-            // try {
-            //   await controller.create(wdto);
-            // } catch (error) {
-            //   expect(error.status).toBe(404);
-            //   expect(error.message).toBe('');
-            // }
+            const wdto = instanceToInstance(dto);
```
Comment (longhoangwkm) [inline src/v1/order-forms/order-forms.controller.spec.ts]: change to let

File: src/v1/order-forms/order-forms.service.ts [hunk 6]
```diff
@@ -19,11 +19,15 @@ export class OrderFormsService {
...
       // check valid time slot from defaultprices
 
-      // check time slot from existed time slot
+      // check time slot from existed time slot of the courtId
 
     }
+    // Create new form
     const order = new OrderForm();
     const sender = await User.findOneBy({id:createOrderFormDto.sender_id});
+    if (sender==null){
```
Comment (longhoangwkm) [inline src/v1/order-forms/order-forms.service.ts]: `if (!sender)` vì undefined cũng nên raise lỗi, nên dùng `===` thay vì `==`

File: src/v1/order-forms/order-forms.controller.spec.ts [hunk 7]
```diff
@@ -0,0 +1,177 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import request from 'supertest';
+import { OrderFormsController } from './order-forms.controller.js';
+import { OrderFormsService } from './order-forms.service.js';
+import {
+  IMPORT_MODULES,
+  clearDB,
```
Comment (longhoangwkm) [inline src/v1/order-forms/order-forms.controller.spec.ts:7]: Refactor clearDB to make it faster

```
export async function clearDB(dataSource: DataSource) {
  const entities = dataSource.entityMetadatas;

  const truncatePromises = entities.map(entity => {
    const repository = dataSource.getRepository(entity.name);
    return repository.query(
      `TRUNCATE "${entity.tableName}" RESTART IDENTITY CASCADE;`
    );
  });

  await Promise.all(truncatePromises);
}
```

File: src/commons/exceptions/InvalidCourt.exception.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
+import { HttpException, HttpStatus } from "@nestjs/common";
+import { error } from "console";
+
+
+export class InvalidCourtException extends HttpException{
+    constructor(message = "invalid_court_exception", data = {}){
```
Comment (longhoangwkm) [inline src/commons/exceptions/InvalidCourt.exception.ts]: Use 2 spaces

**Pull request merged**